wemake_python_styleguide/visitors/ast/loops.py

```diff
@@ -203,6 +206,25 @@ def _check_multiline_loop(self, node: _AnyLoop) -> None:
                 self.add_violation(MultilineLoopViolation(node))
                 break
 
+    def _has_break_raise_or_return(self, node: _AnyLoop) -> bool:
+        for subnode in ast.walk(node):
+            if not isinstance(subnode, ast.While):
+                if isinstance(subnode, ast.Return):
+                    return True
+                if isinstance(subnode, ast.Raise):
+                    return True
+                if isinstance(subnode, ast.Break):
+                    return True
+        return False
+
+    def _check_infinite_while_loop(self, node: _AnyLoop) -> None:
```

Comment:
This can be moved to a another or a new visitor, that is `ast.While` specific.

Reply:
Shouldn't this be inside of this class though, since it says at line 129 "Checks ''for'' and ''while'' loops" ? It seems like _check_multiline_loop also is ast.While specific.

Reply:
I think what I have implemented as of now might work, but let me know!